src/Worker/UpdateServerDirectory.php

```diff
@@ -61,6 +61,9 @@ private static function discoverPoCo(array $gserver)
 
 		$urls = [];
 		foreach ($contacts['entry'] as $entry) {
+			if (empty($entry['urls'])) {
+				continue;
+			}
 			foreach ($entry['urls'] as $url_entry) {
```

Comment:
When you're looking for a specific key in a list of arrays, `array_column` only return existing keys.

```suggestion
		foreach (array_column($contacts['entry'], 'urls') as $urls) {
			foreach ($urls as $url_entry) {
```

Reply:
Okay, I trust you.